src/dialogs/copyPasteOwnership.tsx

```diff
@@ -0,0 +1,139 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+import React, { useEffect, useState } from 'react';
+
+import { AlertVariant } from "@patternfly/react-core/dist/esm/components/Alert";
+import { Button } from "@patternfly/react-core/dist/esm/components/Button";
+import { Form, FormGroup } from "@patternfly/react-core/dist/esm/components/Form";
+import { FormSelect, FormSelectOption } from "@patternfly/react-core/dist/esm/components/FormSelect";
+import { Modal, ModalVariant } from "@patternfly/react-core/dist/esm/components/Modal";
+
+import cockpit from 'cockpit';
+import type { Dialogs, DialogResult } from 'dialogs';
+import { superuser } from 'superuser';
+
+import { useFilesContext } from '../app.tsx';
+import type { ClipboardInfo } from '../app.tsx';
+import { get_owner_candidates } from '../ownership.tsx';
+
+const _ = cockpit.gettext;
+
+async function pasteAsOwner(clipboard: ClipboardInfo, dstPath: string, owner: string,
+    addAlert: (title: string, variant: AlertVariant, key: string, detail?: string) => void) {
+    try {
+        await cockpit.spawn([
+            "cp",
+            "--recursive",
+            ...clipboard.files.map(file => clipboard.path + file.name),
```

Comment:
can you be :100:% sure here that `path` always ends with `/`? If not, this is going to be a mess. My gut feeling is to be defensive and slip a `/` in here -- at worst you get two of them, which doesn't hurt.